trieste/acquisition/multi_objective/partition.py

```diff
@@ -117,25 +179,46 @@ def _bounds_2d(front: TensorType) -> _BoundedVolumes:
 
         return _BoundedVolumes(lower_result, upper_result)
 
-    @classmethod
-    def _divide_conquer_nd(
-        cls, front: TensorType, jitter: float, threshold: TensorType | float = 0
-    ) -> _BoundedVolumes:
-        # Branch and bound procedure to compute the cells covering the non-dominated region.
-        # Generic version: works for an arbitrary number of objectives.
 
-        len_front, number_of_objectives = front.shape
+class DividedAndConquerNonDominated(BoundIndexPartition):
+    """
+    branch and bound procedure algorithm. a divide and conquer method introduced
+    in :cite:`Couckuyt2012`.
+    """
```

Comment:
Just to confirm my understanding, this approach is fast for a few objectives and is used in BoTorch? The other approach is too slow for N>2 but this is a tf issue not an algorithmic issue with the other approach.

Reply:
This `DividedAndConquerNonDominated` is what we have for now: it is just moved into `partition` and this approach has the tf issue (#293), this PR is not containing the new approach (which is faster), the new approach can be referred in #326 and will be prepared until the refractor of the structure change in this PR has been finished.